spec/lib/farscape/plugins_spec.rb

```diff
@@ -223,6 +224,64 @@ def call(env)
 
     end
 
+    context 'extensions' do
+      before(:each) { Farscape.clear }
+      after(:all) { Farscape.clear }
+        
+      let(:entry_point) { "http://localhost:#{RAILS_PORT}"}
+      let(:can_do_hash) { {conditions: 'can_do_anything'} }
+      
+      it 'allows extensions' do
+        module Peacekeeper
+          def pacify!
+            raise 'none shall pass' if enabled_plugins.include?(:Peacekeeper)
+          end
+        end
+        Farscape.register_plugin(name: :Peacekeeper, type: :security, extensions: {Agent: [Peacekeeper]})
+        expect { Farscape::Agent.new.pacify! }.to raise_error('none shall pass')
+      end
+      
+      it 'allows extensions with altering existing methods' do
+        module Peacekeeper
+          def self.extended(base)
+            base.instance_eval do
+              @original_transitions = method(:transitions)
+              def transitions
+                raise 'none shall pass' if @original_transitions.call.keys.include?("drds")
+                @original_transitions.call
+              end
+            end
+          end
+        end
+        Farscape.register_plugin(name: :Peacekeeper, type: :security, extensions: {RepresentorAgent: [Peacekeeper]})
+        expect { Farscape::Agent.new.enter(entry_point).transitions.keys }.to raise_error('none shall pass')
+        expect(Farscape::Agent.new.enter(entry_point).omitting(:Peacekeeper).transitions.keys).to include("drds")
+      end
+      
+      it 'allows discovery extensions' do
+        module ServiceCatalogue
+          def self.extended(base)
+            base.instance_eval do
+              @original_enter = method(:enter)
```

Comment:
I typically see this done using `alias`, but this actually seems better to me, less magical.